### PR TITLE
get the e2e test functional again

### PR DESCRIPTION
**Pull request by ofrobots**
- work around an regression introduced in acorn version 2.6.0
- switch to a promise based test
- remove stale files and dependencies
- Promises. How do they work?

### CODE REVIEW

**Comment by ofrobots** [comment]
@matthewloring PTAL.

File: test/e2e/test.js [hunk 1]
```diff
@@ -20,14 +20,13 @@ function fib(n) {
  * limitations under the License.
  */
 
-process.env.GCLOUD_INSIGHTS_LOGLEVEL=4;
-process.env.GCLOUD_DEBUG_ENABLED=true;
+process.env.GCLOUD_DEBUG_LOGLEVEL=3;
 
 var assert = require('assert');
-var util = require('util');
 var GoogleAuth = require('google-auth-library');
```
Comment (matthewloring) [inline test/e2e/test.js]: Is there a reason why we need the dependency on google-auth-library here? We have corresponding functionality in diagnostics common which should already be used by this package.

Comment (ofrobots) [inline test/e2e/test.js]: Historical baggage. It could be a cleanup task, but isn't essential for getting the test-case functional again.

**Comment by ofrobots** [comment]
For posterity, here's the acorn issue: https://github.com/ternjs/acorn/issues/354

File: test/e2e/test.js [hunk 2]
```diff
@@ -35,118 +34,169 @@ var SCOPES = [
...
+          return a.name === 'n';
+        });
+        assert.ok(arg, 'should find the n argument');
+        assert.strictEqual(arg.value, '10');
+        console.log('Test passed');
+        return true;
+      })
+      .catch(function(err) {
+        console.error(err);
+        process.exit(2);
```
Comment (matthewloring) [inline test/e2e/test.js]: Is there a reason why we catch these and exit with failure as opposed to letting them throw out of the promise?

**Comment by matthewloring** [comment]
Promises :fearful: LGTM with question

File: test/e2e/test.js (hunk 2 above)
Comment (ofrobots) [inline test/e2e/test.js]: I can let it throw out of the promise. I was originally contemplating doing something with the error, but this doesn't seem necessary anymore.

Comment (ofrobots) [inline test/e2e/test.js]: Done.

**Pull request merged**